### ISSUE TITLE
Ability to turn off `_CompileAppManifest` for library projects

### ISSUE THREAD

**Issue by KirillOsenkov** [dotnet/macios#15031]
We are seeing library projects targeting `net6.0-macos` where the target `_CompileAppManifest` and related targets run:
https://github.com/xamarin/xamarin-macios/blob/82482edc70896bfaa7fd63a964d07f070f17baba/msbuild/Xamarin.Shared/Xamarin.Shared.targets#L501

I know it's not a lot of overhead, but my understanding that if we know the libraries definitely have nothing to do with MacOS perhaps we should be able to turn off this machinery?

Perhaps `AppManifestDetectionEnabled` should be respected in a condition in these targets?

See related: 
https://github.com/xamarin/xamarin-macios/issues/13583

I think most of these targets don't need to run for regular projects and we should be able to turn them off?

![image](https://user-images.githubusercontent.com/679326/168370706-f36a1425-9913-4a1e-9a3f-66a743caf11e.png)

**Comment by KirillOsenkov** [comment]
On the other hand, perhaps we should be targeting `net6.0` instead of `net6.0-macos` in our library projects??

**Comment by KirillOsenkov** [comment]
confirmed we need to target net6.0-macos because we use Cocoa APIs from libraries

**Comment by rolfbjarne** [comment]
Yeah, we could look into adding a property that disables app manifest logic for library projects.

**Comment by rolfbjarne** [comment]
This will be fixed with the fix for #19028 (whenever we get to do that).

**Comment by rolfbjarne** [comment]
This was fixed as a side effect of #19028.
